modules/client/jvm/src/main/scala/com/azavea/stac4s/api/client/SearchFilters.scala

```diff
@@ -62,10 +63,10 @@ object SearchFilters extends ClientCodecs {
       filters.bbox,
       filters.datetime,
       filters.intersects,
-      filters.collections,
-      filters.items,
+      filters.collections.some.filter(_.nonEmpty),
+      filters.items.some.filter(_.nonEmpty),
       filters.limit,
-      filters.query,
+      filters.query.some.filter(_.nonEmpty),
```

Comment:
@jisantuc I think this is a better solution - more flixeble codecs rather than the SearchFilters case class definition change.